tests/test_cases/test_cocotb/test_async_coroutines.py

```diff
@@ -135,3 +135,31 @@ async def example():
 
     yield example()
     assert ran
+
+
+@cocotb.test()
+async def test_fork_coroutine_function_exception(dut):
+    async def coro():
+        pass
+
+    try:
```

Comment:
Can you use`with assert_raises` here instead, from `common.py`?

Reply:
Part the check requires converting the `TypeError` exception into string that I can compare against. I could wrap the `try-except` block with `assert_raises` if you still need me to include it.

Reply:
We could slightly change the definition of `assert_raises` to support this.

```python
@contextmanager
def assert_raises(exc_type, pattern=None):
    try:
        yield
    except exc_type:
        if pattern:
            assert re.match(pattern, str(exc_type)), "Correct exception type caught, but message did not match pattern"
    else:
        raise AssertionError("{} was not raised".format(exc_type.__name__))
```

Reply:
I think you mean `except exc_type as e`, `str(e)`, but otherwise that seems like a good idea to me.